### PR TITLE
Fix: Navigation Fixes

### PR DESCRIPTION
**Pull request by nickrandolph**
Adds a new AddProduct page to the Commerce sample to demonstrate full screen navigation
Fix for navigation to get full page navigation to work
Fix for navigation to prevent duplication of event handlers (eg the click handler for Button when Navigation.Request property is set)

### CODE REVIEW

File: src/Uno.Extensions.Navigation.UI/Controls/ActionRequestHandlerBase.cs [hunk 1]
```diff
@@ -3,14 +3,22 @@
...
+
+	protected IRequestBinding? BindAction<TElement, TEventHandler>(
 		TElement view,
 		Func<Action<FrameworkElement>, TEventHandler> eventHandler,
 		Action<TElement, TEventHandler> subscribe,
 		Action<TElement, TEventHandler> unsubscribe
 		)
 		where TElement : FrameworkElement
 	{
+		var _view = view;
```
Comment (jeromelaban) [inline src/Uno.Extensions.Navigation.UI/Controls/ActionRequestHandlerBase.cs:21]: Avoid `_` prefix in method locals

File: src/Uno.Extensions.Navigation.UI/Controls/ActionRequestHandlerBase.cs [hunk 2]
```diff
@@ -87,22 +106,25 @@ public abstract class ActionRequestHandlerBase<TView> : ControlRequestHandlerBas
 			subscribe(view, handler);
 		}
 
-		view.Loaded += (s, e) =>
+		RoutedEventHandler _loadedHandler = (s, e) =>
```
Comment (jeromelaban) [inline src/Uno.Extensions.Navigation.UI/Controls/ActionRequestHandlerBase.cs:109]: avoid `_` prefix in local variables

File: src/Uno.Extensions.Navigation.UI/Controls/SelectorRequestHandler.cs [hunk 3]
```diff
@@ -2,12 +2,13 @@
 
 public class SelectorRequestHandler : ControlRequestHandlerBase<Selector>
 {
-	public override void Bind(FrameworkElement view)
+	public override IRequestBinding? Bind(FrameworkElement view)
 	{
+		var _view = view;
```
Comment (jeromelaban) [inline src/Uno.Extensions.Navigation.UI/Controls/SelectorRequestHandler.cs:7]: Remove `_`

File: src/Uno.Extensions.Navigation.UI/Controls/SelectorRequestHandler.cs [hunk 4]
```diff
@@ -74,13 +75,16 @@ public override void Bind(FrameworkElement view)
 			connect();
 		}
 
-		viewList.Loaded += (s, e) =>
+		RoutedEventHandler _loadedHandler =  (s, e) =>
```
Comment (jeromelaban) [inline src/Uno.Extensions.Navigation.UI/Controls/SelectorRequestHandler.cs:78]: Remove `_`

File: samples/Commerce/Commerce.Shared/Views/AddProductPage.xaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	  Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+
+	<Grid>
+		<Grid.RowDefinitions>
+			<RowDefinition Height="Auto" />
+			<RowDefinition />
+		</Grid.RowDefinitions>
+		<utu:NavigationBar Content="Add Product"
+						   x:Uid="Page_NavigationBar"
+						   utu:AutoLayout.CounterAlignment="Stretch"
```
[the remark this answers is not in the document]
Comment (nickrandolph) [reply samples/Commerce/Commerce.Shared/Views/AddProductPage.xaml]: removed

File: src/Uno.Extensions.Navigation.UI/Controls/ActionRequestHandlerBase.cs (hunk 1 above)
Comment (nickrandolph) [reply src/Uno.Extensions.Navigation.UI/Controls/ActionRequestHandlerBase.cs]: renamed

File: src/Uno.Extensions.Navigation.UI/Controls/SelectorRequestHandler.cs (hunk 4 above)
Comment (nickrandolph) [reply src/Uno.Extensions.Navigation.UI/Controls/SelectorRequestHandler.cs]: renamed

File: src/Uno.Extensions.Navigation.UI/Controls/ActionRequestHandlerBase.cs (hunk 2 above)
Comment (nickrandolph) [reply src/Uno.Extensions.Navigation.UI/Controls/ActionRequestHandlerBase.cs]: renamed

File: src/Uno.Extensions.Navigation.UI/Controls/SelectorRequestHandler.cs (hunk 3 above)
Comment (nickrandolph) [reply src/Uno.Extensions.Navigation.UI/Controls/SelectorRequestHandler.cs]: renamed

File: samples/Commerce/Commerce.Shared/App.xaml.cs [hunk 6]
```diff
@@ -222,7 +222,8 @@ private static void RegisterRoutes(IRouteRegistry routes)
 																						}),
 														new RouteMap("Checkout", View: typeof(CheckoutPage))
 													})
-											})
+											}),
+								new("AddProduct", View: typeof(AddProductPage), ViewModel: typeof(AddProductViewModel)), //, ResultData: typeof(Product)),
```
[the remark this answers is not in the document]
Comment (nickrandolph) [reply samples/Commerce/Commerce.Shared/App.xaml.cs]: removed

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**